src/snowflake/connector/connection.py

```diff
@@ -324,6 +339,7 @@ def __init__(
             setattr(self, f"_{name}", value)
 
         self.heartbeat_thread = None
+        empty_kwargs = not bool(kwargs)
```

Comment:
What's the benefit of calling `bool()` here given that `not some_dict` will always give you a boolean?  Also, I personally like using names like `is_kwargs_empty` for a more descriptive variable name.

Reply:
You know that I always prefer explicit over implicit, I will change it for you though